### ISSUE TITLE
[Bug]: safeQueue goroutine leaked

### ISSUE THREAD

**Issue by w-zr** [matrixorigin/matrixone#6270]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Goroutine leaked after cluster closed.

```
    leak.go:138: Leaked goroutine: goroutine 1242 [select]:
        github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start.func1()
        	/home/wei/matrixone/pkg/vm/engine/tae/logstore/sm/safeq.go:63 +0xfe
        created by github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start
        	/home/wei/matrixone/pkg/vm/engine/tae/logstore/sm/safeq.go:60 +0xea
    leak.go:138: Leaked goroutine: goroutine 1244 [chan receive]:
        github.com/panjf2000/ants/v2.(*Pool).purgePeriodically(0xc00522c850)
        	/home/wei/go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.6/pool.go:69 +0x8b
        created by github.com/panjf2000/ants/v2.NewPool
        	/home/wei/go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.6/pool.go:137 +0x365
```

**Comment by w-zr** [comment]
Leak in `ants.Pool` fixed in #6273.

**Comment by XuPeng-SH** [comment]
fixed by #6273

**Comment by w-zr** [comment]
> fixed by #6273

#6273 only fix leaks in `github.com/panjf2000/ants/v2.(*Pool).purgePeriodically(0xc00522c850)`.
`safeQueue` still leaks.
